### DISCUSSION TITLE
Is there any way to choose a point for tracer BC's?

### DISCUSSION THREAD

**Discussion by Aran55336** [chaos-polymtl/lethe#850] [Q&A]
Hi, I was hoping to make a simulation with a tracer. However, I would like the tracer concentration to be 1 at a chosen point or within a very small sphere at this point, instead of choosing a whole boundary with that concentration. Is there any way to do this?

**Comment by blaisb** [comment]
Dear Aran,
This is absolutely possible. The best approach to achieve this is to set a tracer initial condition. Your post made me realize that this is not documented at the moment, but there is a way to set an initial condition for the tracer.

In the initial conditions sections, you can add a new subsection:

```
subsection tracer
    set Function expression =
end
```

And with this, you can use any mathematical expression you want to set the tracer initial condition. For example, if you want a tracer of 1 left to x=0, you would do something like:
```
subsection tracer
    set Function expression = if(x<0,1,0)
end
```

If you are talking about continuously injecting a tracer at a location (like setting a Dirichlet boundary condition in the domain for the tracer concentration), this is currently not possible, but it would not be a difficult thing to add to the code. You could add a geometric function and force an affice constraint at the location of these nodes for examples. That would be a relatively simple first PR to make and I can help you with that.

Best
Bruno

**Reply by Aran55336** [reply]
Hi Bruno,

I know you had mentioned there's no setting for injecting a tracer at a location. Is there by any chance a setting for continuously injecting a different fluid? I plan on modelling a sparger in a stirred tank so am hoping to model air injection into a viscous fluid.

Best,

Aran

**Reply by blaisb** [reply]
Dear Aran,
Sorry for the huge delay.
Not at the present time no.
